src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java

```diff
@@ -105,6 +112,96 @@ public void checkIfPlanNotOwned(InsurancePlan plan) throws ParseException {
         }
     }
 
+    /**
+     * Adds a claim to the insurance plan of the client.
+     *
+     * @param insurancePlan The insurance plan the claim is to be added to.
+     * @param claim         The claim that is to be added to the insurance plan.
+     * @throws ParseException if the claimId already exists.
+     */
+    public void addClaimToInsurancePlan(InsurancePlan insurancePlan, Claim claim) throws ParseException {
+        if (this.claimIds.contains(claim.getClaimId())) {
+            throw new ParseException(DUPLICATE_CLAIM_ID_MESSAGE);
+        }
+        for (InsurancePlan p : insurancePlans) {
+            if (p.equals(insurancePlan)) {
+                p.claims.add(claim);
+                this.claimIds.add(claim.getClaimId());
+            }
+        }
+    }
+
+    /**
+     * Converts all claims into a string to be saved in JSON file.
+     */
+    public String convertClaimsToJson() {
+        StringBuilder claimsStringBuilder = new StringBuilder();
+        for (InsurancePlan p : this.insurancePlans) {
+            for (Claim c : p.claims) {
+                claimsStringBuilder.append(p.toString())
+                        .append("|")
+                        .append(c.getClaimId())
+                        .append("|")
+                        .append(c.getOpen())
+                        .append("|")
+                        .append(c.getClaimAmount())
+                        .append(",");
+            }
+        }
+
+        if (!claimsStringBuilder.isEmpty()) {
+            claimsStringBuilder.deleteCharAt(claimsStringBuilder.length() - 1);
+        } else {
+            claimsStringBuilder.append("No claims yet.");
+        }
+
+        return claimsStringBuilder.toString();
+    }
+
+    /**
+     * Adds all the claims string from JSON file into their respective insurance plans.
+     *
+     * @param claimsJson String obtained from JSON file of all the claims.
+     * @throws ParseException if there is an error parsing the data from the JSON file.
+     */
+    public void addAllClaimsFromJson(String claimsJson) throws ParseException {
+        String[] claimsString = claimsJson.split(",");
+
+        if (Objects.equals(claimsString[0], "No claims yet.")) {
```

Comment:
why not use `claimsString[0].equals("No claims yet.")` instead?

Reply:
(but also, the name should be changed based on the above comment)

Reply:
yeah, idk why but the IDE suggested that at first. I'll change to urs, makes it more readable